pkg/apiserver/handlers/networkpolicyanalysis/handler.go

```diff
@@ -0,0 +1,145 @@
+	if oldRule.Policy.TierPriority == nil {
+		oldRule.Policy.TierPriority = &effectiveTierPriorityK8sNP
+	}
+	if newRule.Policy.TierPriority == nil {
+		newRule.Policy.TierPriority = &effectiveTierPriorityK8sNP
+	}
+	if oldRule.Policy.TierPriority != newRule.Policy.TierPriority {
+		return *newRule.Policy.TierPriority > *oldRule.Policy.TierPriority
+	}
+	// compare policy priority if tier priority equals
+	if oldRule.Policy.Priority != nil && newRule.Policy.Priority != nil && *newRule.Policy.Priority != *oldRule.Policy.Priority {
+		return *newRule.Policy.Priority > *oldRule.Policy.Priority
+	}
+	if newRule.Policy.Name != oldRule.Policy.Name {
+		return newRule.Policy.Name > oldRule.Policy.Name
+	}
+	// compare rule priority if policy priority equals
+	if newRule.Policy.Rules[newRule.Index].Priority != oldRule.Policy.Rules[oldRule.Index].Priority {
+		return newRule.Policy.Rules[newRule.Index].Priority > oldRule.Policy.Rules[oldRule.Index].Priority
+	}
+	// compare rule index if rule priority equals
+	return newRule.Index < oldRule.Index
+}
+
+// intersectEndpoints returns the predicted rules effective from srcEndpoints to dstEndpoints.
+// Rules returned satisfies either in source applied policies and destination egress rules,
+// or in source ingress rules and destination applied policies.
+func intersectEndpoints(srcEndpoints, dstEndpoints *networkpolicy.EndpointRuleAnalysis) *networkpolicy.Rule {
+	var commonRule *networkpolicy.RuleInfo
+	commonRule = nil
+	for _, rule := range dstEndpoints.EgressRules {
+		if srcEndpoints.Policies.Has(rule.Policy.SourceRef.UID) {
+			if higherPriority(commonRule, rule) {
+				commonRule = rule
+			}
+		}
+	}
+	for _, rule := range srcEndpoints.IngressRules {
+		if dstEndpoints.Policies.Has(rule.Policy.SourceRef.UID) {
+			if higherPriority(commonRule, rule) {
+				commonRule = rule
+			}
+		}
+	}
+	return &networkpolicy.Rule{
+		PolicyRef: networkpolicy.PolicyRef{
+			Namespace: commonRule.Policy.SourceRef.Namespace,
+			Name:      commonRule.Policy.SourceRef.Name,
+			UID:       commonRule.Policy.SourceRef.UID,
+		},
+		Direction: v1beta2.Direction(commonRule.Policy.Rules[commonRule.Index].Direction),
+		RuleIndex: commonRule.Index,
+	}
+}
+
+// HandleFunc creates a http.HandlerFunc which uses an AgentNetworkPolicyInfoQuerier
+// to query network policy rules in current agent.
+func HandleFunc(eq networkpolicy.EndpointQuerier) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		src := r.URL.Query().Get("source")
+		dst := r.URL.Query().Get("destination")
+
+		var srcNS, srcPod, dstNS, dstPod string
+		if src != "" {
+			srcNS, srcPod = ParsePeer(src)
+			if srcPod == "" {
+				http.Error(w, "invalid source format", http.StatusBadRequest)
+				return
+			}
+		}
+		if dst != "" {
+			dstNS, dstPod = ParsePeer(dst)
+			if dstPod == "" {
+				http.Error(w, "invalid destination format", http.StatusBadRequest)
+				return
+			}
+		}
+
```

Comment:
We should validate srcPod and dstPod are both not empty, maybe even at the antctl command level

Reply:
This was handled in the handler, pod name cannot be empty, for namespace similar to traceflow, will default to `default` if empty.